### PR TITLE
Unity test cases are updated for eeprom device

### PR DESCRIPTION
**Pull request by swateeshrivastava**
## Summary
EEPROM test cases are updated as per actual values used by board.

Valid configuration has been taken from the /ec/platform/oc-sdr/cfg/OC_CONNECT_GBC.c for the test cases.
Common register file is created into fake directory.

Function eeprom_write_device_info_record and eeprom_read_device_info_record are only stack holder in the code. No Test case are required from them as of now.

No test case are created for function i2c_eeprom_write and i2c_eeprom_read as they are static function, Can not access them outside the file. Also they are already covered as part of eeprom_reab and eeprom_write. 

## Test Plan
Run Unit Tests (make clean test) and verify that all tests pass including new test cases.
Run firmware build and verify no error is reported.

## Issues
#213

### CODE REVIEW

File: firmware/ec/test/fake/fake_eeprom.c [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    .object_data = &(SX1509_Obj){},
+};
+
+static OcGpio_Pin pin_inven_eeprom_wp = { &gbc_io_0, 2, 32 };
+
+Eeprom_Cfg enable_dev = {
+    .i2c_dev = { 6, 0x45 },
+    .pin_wp = &pin_inven_eeprom_wp,
+    .type = { .page_size = 64, .mem_size = (256 / 8) },
+    .ss = 0,
```
Comment (JoshuaJeyaraj) [inline firmware/ec/test/fake/fake_eeprom.c]: can we replace SS number with the equivalent macros which are used in code .

File: firmware/ec/platform/oc-sdr/schema/auto_schema.c [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-* This source code is licensed under the BSD-style license
-* found in the LICENSE file in the root directory of this
-* source tree. An additional grant of patent rights can be
-* found in the PATENTS file in the same directory.
-*
-* WARNING: Do not modify this file by hand.  It is auto
-*          generated from the json schema definition.
-*          Refer to sdtester.py
-*/
+ * Copyright (c) 2018-present, Facebook, Inc.
```
[the remark this answers is not in the document]
Comment (jspectacular) [reply firmware/ec/platform/oc-sdr/schema/auto_schema.c]: If you want to modify spacing, the correct file to modify is ~firmware/utilities/schema/schemautils.py, writeSchema(...).

[the remark this answers is not in the document]
Comment (jspectacular) [reply firmware/ec/platform/oc-sdr/schema/auto_schema.c]: Make the changes there, rebuild, and `git add` modified schemautils.py and auto_schema.c

File: firmware/ec/src/devices/eeprom.c [hunk 3]
```diff
@@ -50,13 +50,16 @@ static ReturnStatus i2c_eeprom_read(I2C_Handle i2cHandle,
...
+        return false;
+    }
     /* Configure our WP pin (if any) and set to be low (protected) by default */
     if (cfg->pin_wp) {
         OcGpio_configure(cfg->pin_wp, OCGPIO_CFG_OUTPUT | OCGPIO_CFG_OUT_HIGH);
     }
 
     /* Test communication to the EEPROM */
-    uint8_t test_byte;
+    uint16_t test_byte;
```
[the remark this answers is not in the document]
Comment (swateeshrivastava) [reply firmware/ec/src/devices/eeprom.c]: This is related to below issue.
https://github.com/markhor/OpenCellular/issues/465

File: firmware/ec/src/devices/eeprom.c [hunk 4]
```diff
@@ -225,6 +234,9 @@ static ReturnStatus i2c_eeprom_read(I2C_Handle i2cHandle, uint16_t slaveAddress,
  *****************************************************************************/
 ReturnStatus eeprom_disable_write(Eeprom_Cfg *cfg)
 {
+    if ((cfg == NULL) || (cfg->pin_wp == NULL)) {
```
[the remark this answers is not in the document]
Comment (swateeshrivastava) [reply firmware/ec/src/devices/eeprom.c]: Done.

File: firmware/ec/src/devices/eeprom.c [hunk 5]
```diff
@@ -50,13 +50,16 @@ static ReturnStatus i2c_eeprom_read(I2C_Handle i2cHandle,
  *****************************************************************************/
 bool eeprom_init(Eeprom_Cfg *cfg)
 {
+    if ((cfg == NULL) || (cfg->pin_wp == NULL)) {
```
[the remark this answers is not in the document]
Comment (swateeshrivastava) [reply firmware/ec/src/devices/eeprom.c]: Done

File: firmware/ec/test/fake/fake_eeprom.c (hunk 1 above)
Comment (swateeshrivastava) [reply firmware/ec/test/fake/fake_eeprom.c]: Done

**Pull request merged**